tools/Azure.Mcp.Tools.Cosmos/src/Validation/CosmosQueryValidator.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.RegularExpressions;
+using Azure.Mcp.Core.Exceptions;
+
+namespace Azure.Mcp.Tools.Cosmos.Validation;
+
+/// <summary>
+/// Lightweight validator to reduce risk of executing unsafe queries via the Cosmos SQL query tool.
+/// Cosmos SQL syntax is inherently read-only (SELECT only), but this validator blocks:
+/// 1. Multiple/stacked statements that could bypass security
+/// 2. Comments that could hide malicious code
+/// 3. Attempts to execute stored procedures/triggers (which CAN modify data)
+/// 4. Common SQL injection patterns
+/// Note: Stored procedures and triggers are executed via SDK APIs, not SQL queries, so they cannot
+/// be invoked through this query interface. This is defense-in-depth validation.
+/// </summary>
+internal static class CosmosQueryValidator
```

Comment:
Just comparing this with the one we added for Postgres (https://github.com/microsoft/mcp/blob/main/tools/Azure.Mcp.Tools.Postgres/src/Validation/SqlQueryValidator.cs) - is there a reason the Cosmos validation looks slightly different? If not, should we align the logic here as well?

Reply:
You are right. Cosmos is different from Postgre.

CosmosDB SQL has only SELECT statements based on the grammar. And we only need to block of patterns of stored procedures and triggers. https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/how-to-write-stored-procedures-triggers-udfs?tabs=javascript.

Reply:
thanks for the clarification Xiang, looks great to me